### ISSUE TITLE
version 5.0 does not start

### ISSUE THREAD

**Issue by shantanuo** [docker-library/elasticsearch#114]
I am not able to start elasticsearch version 5.0 using docker.

```
# docker run -d -v /tmp/:/usr/share/elasticsearch/config  -p 9200:9200 -p 9300:9300 elasticsearch:5.0.0-alpha5
8bdc08ee04497548c4e57d0e8ee075359bc95eafa3e1814b1c448f56597fff59

# docker logs 8bdc08ee04497548c4e57d0e8ee075359bc95eafa3e1814b1c448f56597fff59
log4j:WARN No appenders could be found for logger (bootstrap).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
```

The same image works with default tag.
_____

**Update:**

When I tried this command, I get an error...

docker run -d --name elasticsearch -p 9200:9200 -p 9300:9300 -e ES_JAVA_OPTS="-Xms512m -Xmx512m" elasticsearch:5

```
[2016-08-10 15:38:11,815][INFO ][plugins                  ] [YL2O1bL] no plugins loaded
[2016-08-10 15:38:14,412][INFO ][node                     ] [YL2O1bL] initialized
[2016-08-10 15:38:14,413][INFO ][node                     ] [YL2O1bL] starting ...
[2016-08-10 15:38:14,561][INFO ][transport                ] [YL2O1bL] publish_address {172.17.0.2:9300}, bound_addresses {[::]:9300}
[2016-08-10 15:38:14,568][INFO ][bootstrap                ] [YL2O1bL] bound or publishing to a non-loopback or non-link-local address, enforcing bootstrap checks
[2016-08-10 15:38:14,574][WARN ][bootstrap                ] [] uncaught exception in thread [main]
org.elasticsearch.bootstrap.StartupError: java.lang.RuntimeException: bootstrap checks failed
max virtual memory areas vm.max_map_count [65530] likely too low, increase to at least [262144]
        at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:105)
        at org.elasticsearch.bootstrap.Elasticsearch.execute(Elasticsearch.java:96)
        at org.elasticsearch.cli.SettingCommand.execute(SettingCommand.java:54)
        at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:88)
        at org.elasticsearch.cli.Command.main(Command.java:54)
        at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:75)
        at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:68)
Caused by: java.lang.RuntimeException: bootstrap checks failed
max virtual memory areas vm.max_map_count [65530] likely too low, increase to at least [262144]
        at org.elasticsearch.bootstrap.BootstrapCheck.check(BootstrapCheck.java:132)
        at org.elasticsearch.bootstrap.BootstrapCheck.check(BootstrapCheck.java:85)
        at org.elasticsearch.bootstrap.BootstrapCheck.check(BootstrapCheck.java:65)
        at org.elasticsearch.bootstrap.Bootstrap$5.validateNodeBeforeAcceptingRequests(Bootstrap.java:178)
        at org.elasticsearch.node.Node.start(Node.java:471)
        at org.elasticsearch.bootstrap.Bootstrap.start(Bootstrap.java:193)
        at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:257)
        at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:101)
        ... 6 more
        Suppressed: java.lang.IllegalStateException: max virtual memory areas vm.max_map_count [65530] likely too low, increase to at least [262144]
                at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
                at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1374)
                at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
                at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
                at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
                at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
                at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
                at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
                at org.elasticsearch.bootstrap.BootstrapCheck.check(BootstrapCheck.java:133)
                ... 13 more
[2016-08-10 15:38:14,582][INFO ][node                     ] [YL2O1bL] stopping ...
[2016-08-10 15:38:14,616][INFO ][node                     ] [YL2O1bL] stopped
[2016-08-10 15:38:14,616][INFO ][node                     ] [YL2O1bL] closing ...
[2016-08-10 15:38:14,649][INFO ][node                     ] [YL2O1bL] closed

```

**Comment by shantanuo** [comment]
I found the solution when I read the other issue...
Figure out what's up with 5.0 #98

I need to run this on host:
sysctl -w vm.max_map_count=262144

This should be clearly documented somewhere on home page.

**Comment by yosifkit** [comment]
Considering that 5.0 is an alpha release I don't think we should add documentation about special running considerations until it is a full release.  Closing as duplicate of #98.

**Comment by ip2k** [comment]
FWIW another suggestion on the other bug worked for me:

`-e ES_JAVA_OPTS="-Xms1g -Xmx1g"` as in `docker run -e ES_JAVA_OPTS="-Xms1g -Xmx1g" -v "$PWD/esdata":/usr/share/elasticsearch/data elasticsearch:5`

**Comment by afrazkhan** [comment]
Now that you have released 5.0, please document this so that somebody else doesn't pull their hair out for over a day about it. The worst thing about this is that if you start it differently (from an init script for example), you get a silent exit. No logs, no errors, nothing.

**Comment by tianon** [comment]
@afrazkhan indeed, see https://github.com/docker-library/docs/issues/734 (where we added this to the documentation) :+1:
